samcli/commands/deploy/command.py

```diff
@@ -287,13 +294,23 @@ def do_cli(
             config_file=config_file,
         )
         guided_context.run()
-    elif resolve_s3 and bool(s3_bucket):
-        raise DeployResolveS3AndS3SetError()
-    elif resolve_s3:
-        s3_bucket = manage_stack(profile=profile, region=region)
-        click.echo(f"\n\t\tManaged S3 bucket: {s3_bucket}")
-        click.echo("\t\tA different default S3 bucket can be set in samconfig.toml")
-        click.echo("\t\tOr by specifying --s3-bucket explicitly.")
+    else:
+        if resolve_s3:
+            if bool(s3_bucket):
+                raise DeployResolveS3AndS3SetError()
+            s3_bucket = manage_stack(profile=profile, region=region)
```

Comment:
I wonder if we should encapsulate creation of other resources other than template to its own class, that way addition of new resources becomes easier in the future.

Reply:
This is a good idea, but current the difference in behavior between package and deploy regarding having --resolve-image-repos will make it less beneficial. We can include this refactoring in the follow up implementation of adding --resolve-image-repos to package.

Reply:
could we add a todo here.